### PR TITLE
Implement event log in the ledger

### PR DESCRIPTION
**Pull request by sug0**
Based on https://github.com/anoma/namada/pull/587

Implements the second task of issue https://github.com/anoma/namada/issues/563

### CODE REVIEW

**Review by batconjurer — APPROVED** [review]

File: apps/src/lib/node/ledger/events/log/dumb_queries.rs
Comment (james-chf) [inline apps/src/lib/node/ledger/events/log/dumb_queries.rs:1]: Will we still need `dumb_queries.rs` or use the approach of having `Path` `Accepted`/`Applied` enum variants you mentioned? (in https://github.com/anoma/namada/blob/a669ca573f1d3f61b87cdcb8eba7cb9a6f872a6a/apps/src/lib/node/ledger/rpc.rs)

Comment (sug0) [reply apps/src/lib/node/ledger/events/log/dumb_queries.rs:1]: they're still needed, but I added two new methods to directly instantiate a matcher cd6746a4f31fa620495c7b1c0c846df12ee3a1ac

File: apps/src/lib/node/ledger/events/log.rs
Comment (james-chf) [inline apps/src/lib/node/ledger/events/log.rs:59]: I still think this should accept `IntoIterator<Item = Event>` since we always clone the borrowed events - the caller can clone as necessary. It would make it explicit this fn actually needs its own copy of all the events passed in.

File: apps/src/lib/node/ledger/events/log.rs
Comment (james-chf) [inline apps/src/lib/node/ledger/events/log.rs:96]: nit
```suggestion
            .filter(move |&event| matcher.matches(event))
```

File: apps/src/lib/node/ledger/events/log/dumb_queries.rs
Comment (james-chf) [inline apps/src/lib/node/ledger/events/log/dumb_queries.rs:64]: Re. https://github.com/anoma/namada/pull/607#discussion_r996798254 do we need the regex matching anymore? We can directly construct the matchers now (could even just be called `EventMatcher` or smth)

**Review by james-chf — CHANGES_REQUESTED** [review]

File: apps/src/lib/node/ledger/events/log/dumb_queries.rs
Comment (sug0) [reply apps/src/lib/node/ledger/events/log/dumb_queries.rs:64]: I can remove this in the next PR https://github.com/anoma/namada/pull/610

File: apps/src/lib/node/ledger/events/log/dumb_queries.rs
Comment (sug0) [reply apps/src/lib/node/ledger/events/log/dumb_queries.rs:45]: marking this as solved, since I'll remove query parsing

File: apps/src/lib/node/ledger/events/log.rs
Comment (sug0) [reply apps/src/lib/node/ledger/events/log.rs:82]: idem

**Comment by sug0** [comment]
the PoS VP state machine proptest failed. I wonder if this might be because our eth secp logic isn't integrated into the VP in this branch, yet

**Comment by james-chf** [comment]
pls update wasm

**Review by james-chf — APPROVED** [review]
lgtm to merge once CI passes (PoS test I guess is a flake then)

**Comment by james-chf** [comment]
(or wasm is updated now so CI should pass on `eth-bridge-integration` after this merges I think anyway)

**Pull request merged**
